cloud/pkg/dynamiccontroller/application/listener.go

```diff
@@ -52,7 +53,15 @@ func (l *SelectorListener) sendObj(event watch.Event, messageLayer messagelayer.
 	}
 
 	msg.SetResourceVersion(accessor.GetResourceVersion())
-	resource, err := messagelayer.BuildResource(l.nodeName, accessor.GetNamespace(), l.gvr.Resource, accessor.GetName())
+	name := accessor.GetName()
+	if name == "" {
+		name = v2.NullName
+	}
+	namespace := accessor.GetNamespace()
+	if namespace == "" {
+		namespace = v2.NullNamespace
+	}
```

Comment:
Looks good, the namespace can be null for clusterscope object, but name may impossible to be null? :)

Reply:
With this patch, flannel can run in the edge?

Reply:
You are right. name not be null. And with this patch.flannel should be run in edge.

Reply:
> And with this patch.flannel should be run in edge.

Wow!! Great! 

> name not be null.

But I think we can keep the name related logic here?

Reply:
> But I think we can keep the name related logic here?

Both are ok.(◐‿◑)﻿